CMakeLists.txt

```diff
@@ -1,84 +1,93 @@
-cmake_minimum_required(VERSION 2.8.3)
+cmake_minimum_required(VERSION 3.5)
 project(urg_node)
 
-find_package(catkin REQUIRED COMPONENTS
-  diagnostic_updater dynamic_reconfigure laser_proc message_generation
-  nodelet rosconsole roscpp sensor_msgs std_msgs std_srvs tf urg_c
-)
-set_directory_properties(PROPERTIES COMPILE_OPTIONS "-std=c++11")
+# Default to C++14
+if(NOT CMAKE_CXX_STANDARD)
+  set(CMAKE_CXX_STANDARD 14)
+endif()
+if(CMAKE_COMPILER_IS_GNUCXX OR CMAKE_CXX_COMPILER_ID MATCHES "Clang")
+  add_compile_options(-Wall -Wextra -Wpedantic)
+endif()
 
-# Dynamic reconfigure support
-generate_dynamic_reconfigure_options(cfg/URG.cfg)
+find_package(ament_cmake REQUIRED)
+find_package(Boost COMPONENTS thread REQUIRED)
```

Comment:
This is kind of weird, since we don't use `Boost::thread` anywhere.  But there doesn't seem to be a component for `crc`; it just sort of comes in as part of Boost.  That deserves at least a comment until we can remove this lingering boost dependency.